modules/core/src/main/java/org/apache/ignite/internal/managers/systemview/ScanQuerySystemView.java

```diff
@@ -146,7 +148,13 @@ public QueryDataIterator() {
 
                     reqMap = next.getValue();
 
-                    qriesIter = next.getValue().entrySet().iterator();
+                    Set<Map.Entry<Long, GridFutureAdapter<GridCacheQueryManager.QueryResult<K, V>>>> cpOfEntries;
+                    // It should prevent concurrent modification of the map while a snapshot of the current state is in progress
+                    synchronized (reqMap) {
+                        cpOfEntries = new LinkedHashSet(reqMap.entrySet());
+                    }
+
+                    qriesIter = cpOfEntries.iterator();
```

Comment:
```suggestion
                    Map<Long, GridFutureAdapter<GridCacheQueryManager.QueryResult<K, V>>> cpOfEntries;
                    // It should prevent concurrent modification of the map while a snapshot of the current state is in progress
                    synchronized (reqMap) {
                        cpOfEntries = new HashMap(reqMap);
                    }

                    qriesIter = cpOfEntries.entrySet().iterator();
```

Reply:
No, we have ordered a Set and should keep ordered one after copy